src/components/forms/Combobox.tsx

```diff
@@ -580,6 +580,8 @@ export function ComboboxStatic<T = any>({
                   if (onDismiss) {
                     e.preventDefault();
 
+                    setQuery('');
```

Comment:
@beganovich Clearing the query by setting it to an empty string allows the user to obtain refreshed results without any filters when they click the X button.